lib/rex/post/smb/ui/console.rb

```diff
@@ -97,6 +97,10 @@ def run_command(dispatcher, method, arguments)
             log_error(e.message)
           rescue ::Errno::EPIPE, ::OpenSSL::SSL::SSLError, ::IOError
             session.kill
+          rescue ::RubySMB::Error::EncryptionError => e
```

Comment:
I don't think an encryption error equates to the server no longer being alive - is there  any other exceptions that we can catch here maybe 👀

Reply:
I opened a PR in [ruby_smb](https://github.com/rapid7/ruby_smb/pull/263) to now raise an `EOFError` instead if the response is `nil`.